### PR TITLE
Fix GUI crash when filtering by UUID on Postgres

### LINKED ISSUE #1084: Crash in Pepys-Admin
I'm trying to find a platform using platform ID using `Maintain data`

I selected `Platforms`,  then the `Platform_ID` field, and the `Like` operator. As I start to enter the start of the platform id `e957` I get this error:
image

### PR DESCRIPTION
**Pull request by robintw**
## 🧰 Issue
Fixes #1084 

## 🚀 Overview: 
Fixes crash when trying to search by a field with a UUID type in the Maintenance GUI.

## 🤔 Reason: 
Stop Pepys crashing

## 🔨Work carried out:

- [x] Ensure that both SQLite _and_ Postgres UUID types are checked for before casting to a string when adding a string filter on a UUID type
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR, adding `interactive_review` if reviewers will need to see UI
- [x] I have extended/updated the documentation in `\docs` folder
- [x] Any database content changes (Create, Edit, Delete) are recorded in the Log/Changes tables
- [x] Any database schema changes are implemented via `alembic revision` [transitions](https://pepys-import.readthedocs.io/en/latest/database_migration.html#how-to-use-it-for-developers)
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

## 📝 Developer Notes:
When doing a `LIKE` query on a UUID field, we need to convert it to a string first. In the code that generates the SQLAlchemy filter conditions, this was only happening for SQLite UUID types (which are basically just a string anyway), but not for Postgres UUID types. This fixes that.

### CODE REVIEW

File: pepys_admin/maintenance/widgets/filter_widget_utils.py [hunk 1]
```diff
@@ -122,7 +123,7 @@ def handle_three_variables(class_obj, output, idx, query_list, data_store):
 
     if ops == "LIKE":
         try:
-            if isinstance(col.type, UUIDType):
+            if isinstance(col.type, UUIDType) or isinstance(col.type, UUID):
```
Comment (IanMayo) [inline pepys_admin/maintenance/widgets/filter_widget_utils.py:126]: Could you explain what the issue was here, please @robintw ?

Comment (robintw) [reply pepys_admin/maintenance/widgets/filter_widget_utils.py]: I mentioned it briefly in the Developer Notes in the PR description, but basically this if statement is checking whether the column is a UUID. If it is, then it casts it to string before trying to do a `LIKE` operation on it. However, we were only checking whether it was a `UUIDType` (which is the type used for UUIDs on SQLite), and not checking against `UUID` (which is the type used for UUIDs on Postgres). Therefore, it failed on Postgres but worked on SQLite.

**Review by IanMayo — APPROVED** [review]
Tested. Filtering by platform_id works

File: pepys_admin/maintenance/widgets/filter_widget_utils.py (hunk 1 above)
Comment (IanMayo) [reply pepys_admin/maintenance/widgets/filter_widget_utils.py]: Aah, got you - thanks.

**Review by IanMayo — APPROVED** [review]
Tested, working fine

**Pull request merged**